pkg/nsx/services/vpc/builder.go

```diff
@@ -28,16 +28,20 @@ func generateIPBlockKey(block model.IpAddressBlock) string {
 }
 
 func buildNSXVPC(obj *v1alpha1.NetworkInfo, nsObj *v1.Namespace, nc common.VPCNetworkConfigInfo, cluster string,
-	nsxVPC *model.Vpc, useAVILB bool) (*model.Vpc,
+	nsxVPC *model.Vpc, useAVILB bool, lbProviderChanged bool) (*model.Vpc,
 	error) {
 	vpc := &model.Vpc{}
 	if nsxVPC != nil {
 		// for upgrade case, only check public/private ip block size changing
-		if !IsVPCChanged(nc, nsxVPC) {
+		if !IsVPCChanged(nc, nsxVPC) && !lbProviderChanged {
 			log.Info("no changes on current NSX VPC, skip updating", "VPC", nsxVPC.Id)
 			return nil, nil
 		}
 		// for updating vpc case, use current vpc id, name
+		if useAVILB && lbProviderChanged {
```

Comment:
no need to check lbProviderChanged, if useAVILB is true, then enable loadBalancerVPCEndpointEnabled?

Reply:
if the lbProviderChanged is false, that means is AVILB and loadBalancerVPCEndpointEnabled is true.